main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -421,6 +421,25 @@ object WeederError {
     })
   }
 
+  /**
+    * An error raised to indicate that a ForA-loop contains other ForFragments than Generators.
+    *
+    * @param loc the location of the for-loop in which the for-fragment appears.
+    */
+  case class IllegalForAFragment(loc: SourceLocation) extends WeederError with Recoverable {
```

Comment:
Why is this illegal?

Reply:
This was how it was parsed before this PR, so I just wanted to make sure that we kept the same restrictions on it until we decided otherwise. So the error can certainly be removed in another PR.

Reply:
Let's open a ticket for discussion